subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r56/ToolingApiEclipseModelTestSourcesCrossVersionSpec.groovy

```diff
@@ -67,4 +68,84 @@ class ToolingApiEclipseModelTestSourcesCrossVersionSpec extends ToolingApiSpecif
     private boolean hasTestAttributes(EclipseClasspathEntry entry) {
         entry.classpathAttributes.find { it.name == 'test' && it.value == 'true' }
     }
+
+    def "can use compile classpath for API and implementation separation"() {
```

Comment:
❌ The method probably need to have the `@TargetGradleVersion(">=6.1")` annotation. Also, I think this test should be in the `org.gradle.plugins.ide.tooling.r61` package.

Reply:
My intention was to show that you can make this work with the workaround (see `if (targetVersion.baseVersion < GradleVersion.version('6.1'))` further down) for the last 5.x release. Because it might be interesting for users on 5. And it is what I commented here: https://github.com/eclipse/buildship/issues/815#issuecomment-565408634

(This test is passing with all versions 5.6+)

Reply:
👍